plugins/module_utils/foreman_helper.py

```diff
@@ -245,12 +246,12 @@ def check_requirements(self):
         if not HAS_APYPIE:
             self.fail_json(msg='The apypie Python module is required', exception=APYPIE_IMP_ERR)
 
-    @_exception2fail_json(msg="Failed to connect to Foreman server: %s ")
+    @_exception2fail_json(msg="Failed to connect to Foreman server: {0}")
     def connect(self):
         self.foremanapi = apypie.Api(
             uri=self._foremanapi_server_url,
             username=self._foremanapi_username,
```

Comment:
IMHO the username also should be `to_bytes`

Reply:
What about the url then? I think, DNS knows about unicode...

Reply:
don't go to far ;)